### ISSUE TITLE
Messaging system somtimes panics from 'invalid memory address or nil pointer dereference'.

### ISSUE THREAD

**Issue by evanlinjin** [skycoin/skywire-testnet#336]
Extracts from logs provided by @ayuryshev 
```
[2019-05-08T13:30:24+03:00] INFO [messenger]: Opened new link with the server 035915c609f71d0c7df27df85ec698ceca0cb262590a54f732e3bbd0cc68d89282
[2019-05-08T13:30:24+03:00] INFO [messenger]: Updating discovery entry
[2019-05-08T13:30:24+03:00] INFO [messenger]: Opened new channel local ID 0, remote ID 0 with 024ec47420176680816e0406250e7156465e4531f5b26057c9f6297bb0303558c7
[2019-05-08T13:30:24+03:00] WARN [messenger]: Failed to send frame to 035915c609f71d0c7df27df85ec698ceca0cb262590a54f732e3bbd0cc68d89282: noise handshake: chacha20poly1305: message authentication failed
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x9baf16]

goroutine 133 [running]:
github.com/flynn/noise.(*CipherState).Encrypt(...)
	/home/isk/go/pkg/mod/github.com/flynn/noise@v0.0.0-20180327030543-2492fe189ae6/state.go:33
github.com/skycoin/skywire/internal/noise.(*Noise).EncryptUnsafe(0xc0002126c0, 0xc000482000, 0x208, 0x30a, 0xc00022a618, 0x0, 0xc000084ea0)
	/home/isk/github.com/skycoin/src/skywire/internal/noise/noise.go:126 +0xe6
github.com/skycoin/skywire/pkg/messaging.(*channel).Write.func1(0xc000224990, 0xc000482000, 0x208, 0x30a, 0xc00044d2c0)
	/home/isk/github.com/skycoin/src/skywire/pkg/messaging/channel.go:99 +0xe0
created by github.com/skycoin/skywire/pkg/messaging.(*channel).Write
	/home/isk/github.com/skycoin/src/skywire/pkg/messaging/channel.go:95 +0x17d
```
```
[2019-05-08T13:31:20+03:00] INFO [messenger]: Opened new link with the server 035915c609f71d0c7df27df85ec698ceca0cb262590a54f732e3bbd0cc68d89282
[2019-05-08T13:31:20+03:00] INFO [messenger]: Updating discovery entry
[2019-05-08T13:31:21+03:00] INFO [messenger]: Opened new channel local ID 0, remote ID 0 with 031b80cd5773143a39d940dc0710b93dcccc262a85108018a7a95ab9af734f8055
[2019-05-08T13:31:21+03:00] WARN [messenger]: Failed to send frame to 035915c609f71d0c7df27df85ec698ceca0cb262590a54f732e3bbd0cc68d89282: noise handshake: chacha20poly1305: message authentication failed
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x9baf16]

goroutine 208 [running]:
github.com/flynn/noise.(*CipherState).Encrypt(...)
	/home/isk/go/pkg/mod/github.com/flynn/noise@v0.0.0-20180327030543-2492fe189ae6/state.go:33
github.com/skycoin/skywire/internal/noise.(*Noise).EncryptUnsafe(0xc0001d4540, 0xc00011a000, 0x208, 0x2e0, 0xc00022a0c0, 0xc00011c4c0, 0xc000441440)
	/home/isk/github.com/skycoin/src/skywire/internal/noise/noise.go:126 +0xe6
github.com/skycoin/skywire/pkg/messaging.(*channel).Write.func1(0xc0002dc120, 0xc00011a000, 0x208, 0x2e0, 0xc000119800)
	/home/isk/github.com/skycoin/src/skywire/pkg/messaging/channel.go:99 +0xe0
created by github.com/skycoin/skywire/pkg/messaging.(*channel).Write
	/home/isk/github.com/skycoin/src/skywire/pkg/messaging/channel.go:95 +0x17d
```

**Comment by evanlinjin** [comment]
The cause of this problem seems to be, that the `noise.CipherState` is deleted before `(noise.Noise).EncryptUnsafe` is called.

![image](https://user-images.githubusercontent.com/14065705/57373157-c3ef3d80-71eb-11e9-9ac2-b92a613f4320.png)

**Comment by jdknives** [comment]
Closed by #337
